src/main/java/org/hl7/davinci/priorauth/ClaimEndpoint.java

```diff
@@ -599,7 +606,36 @@ private Bundle generateAndStoreClaimResponse(Bundle bundle, Claim claim, String
 
     @Override
     public void run() {
-      updateClaim(bundle, claimId, patient, disposition);
+      if (updateClaim(bundle, claimId, patient, disposition) != null) {
+        // Check for subscription
+        Map<String, Object> constraintMap = new HashMap<String, Object>();
+        constraintMap.put("claimResponseId", claimId);
+        constraintMap.put("patient", patient);
+        Subscription subscription = (Subscription) App.getDB().read(Database.SUBSCRIPTION, constraintMap);
+
+        // Send notification to the subscriber
+        if (subscription != null) {
+          SubscriptionChannelType subscriptionType = subscription.getChannel().getType();
```

Comment:
Should we have error handling around the SubscriptionChannelType when the subscription is created like you do here?

Reply:
The error handling "like you do here" referring to checking the type?

Reply:
Yes, we should check the type when it comes in and send an error back then.

Reply:
Ah yes I agree

Reply:
Updated to do error handling in subscription endpoint